src/interfaces/repository.interface.ts

```diff
@@ -0,0 +1,9 @@
+interface Repository {
+  findAll(): Promise<any>;
+  save(resource: any): Promise<any>;
+  update(id: string, resource: any): Promise<any>;
+  delete(id: string): Promise<any>;
+  toggleDone(id: string): Promise<any>;
+}
+
+export default Repository;
```

Comment:
Très bonne idée mais j'aurais tiré parti de la généricité plutôt que de mettre du `any` partout (on n'est pas en javascript et le mot clef `any` est vraiment à réserver à des cas très limités, il vaut mieux utiliser `unknown` que `any` parce que typescript te forcera alors à vérifier les types avant de faire des opérations).

```suggestion
interface Repository<T> {
  findAll(): Promise<T>;
  save(resource: Partial<T>): Promise<T>;
  update(id: string, resource: Partial<T>): Promise<T>;
  delete(id: string): Promise<void>;
  toggleDone(id: string): Promise<T>;
}

export default Repository;
```